### PR TITLE
Command to move projects amongst users

### PR DESCRIPTION
**Pull request by bmarika**
Fixes https://github.com/onaio/onadata/issues/1491

Carried Over from #1491 
Transfer projects to a different user/organisation account.
- [x] Reassign created_by and organisation to the new owner/organisation.
- [x] Reassign all XForms under the user or project to the new
      user/organisation. Updating the fields `user` and `created_by`.
 - [x] Reassign all MergedXForms under the user or project to the new
      user/organisation. 
- [x] Reassign all Dataviews under the user or project to the new
      user/organisation. 
- [x] Apply project and form permissions to the new user.
- [x] Remove project and form permissions from the old user?

### CODE REVIEW

File: onadata/apps/logger/management/commands/transfer_project.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        xforms = XForm.objects.filter(project=project)
+        for form in xforms:
+            form.user = user
+            form.created_by = user
+            form.save()
+            self.update_data_viewws(form)
+            set_project_perms_to_xform(form, project)
+
+    @staticmethod
+    def update_data_viewws(form):
```
Comment (ukanga) [inline onadata/apps/logger/management/commands/transfer_project.py]: Is the spelling a typo or intentional? `update_data_viewws`? I would suggest we stick to `update_data_views`.

File: onadata/apps/logger/management/commands/transfer_project.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        except user_model.DoesNotExist:
+            self.errors.append("User {0} does not exist \n".format(username))
+        return user
+
+    def update_xform_with_new_user(self, project, user):
+        """
+        Update XForm user update the DataViews and also set the permissions
+        for the xform and the project.
+        """
+        xforms = XForm.objects.filter(project=project)
```
Comment (ukanga) [inline onadata/apps/logger/management/commands/transfer_project.py]: We may need to address the issue for inactive forms i.e `downloadable=False`, there is usually an exception that is raised for inactive forms. I am not sure if it applies to the other models. Also, for the situation where an object has been soft deleted.

File: onadata/apps/logger/management/commands/transfer_project.py (hunk 1 above)
Comment (bmarika) [reply onadata/apps/logger/management/commands/transfer_project.py]: It is a typo. Will fix it.

File: onadata/apps/logger/management/commands/transfer_project.py (hunk 2 above)
Comment (bmarika) [reply onadata/apps/logger/management/commands/transfer_project.py]: I have made changes to address both scenarios.

File: onadata/apps/logger/management/commands/transfer_project.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    def update_merged_xform(project, user):
+        merged_xforms = MergedXForm.objects.filter(
+            project=project, deleted_at__isnull=True)
+        for form in merged_xforms:
+            form.user = user
+            form.created_by = user
+            form.save()
+            set_project_perms_to_xform(form, project)
+
+    def update_enketo_urls(self, form):
```
Comment (ukanga) [inline onadata/apps/logger/management/commands/transfer_project.py]: I think this does correctly create Enketo metadata records for a new Enketo URI from the move but maintains the old URI. I don't think this sufficiently addresses the issue in #845. A fix for that could be having an endpoint that does not include user/org identifying information in the URI.

File: onadata/apps/logger/management/commands/transfer_project.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+from django.db import transaction
+
+from onadata.apps.logger.models import Project, XForm, DataView, MergedXForm
+from onadata.apps.logger.models.project import set_object_permissions \
+    as set_project_permissions
+from onadata.libs.utils.project_utils import set_project_perms_to_xform
+
+
+class Command(BaseCommand):  # pylint: disable=C0111
+    help = 'A command to reassign a project form one user to the other.'
```
Comment (ukanga) [inline onadata/apps/logger/management/commands/transfer_project.py]: This command transfers all projects from one user to another user, the description seems to indicate we are only transferring one project.

Could we have two options:
1. `--project` -  the project id of the project to transfer
2. `--all` - transfer all projects

File: onadata/apps/logger/management/commands/transfer_project.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    errors = []
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--current_owner',
+            help='Username of the current owner of of the projects',
+        )
+        parser.add_argument(
+            '--new_owner',
+            help='TUsername of new owner of the projects',
```
Comment (ukanga) [inline onadata/apps/logger/management/commands/transfer_project.py]: I think this could be a typo `TUsername`.

**Pull request merged**